subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -2667,11 +2745,23 @@ void lwm2m_udp_receive(struct lwm2m_ctx *client_ctx, uint8_t *buf, uint16_t buf_
 			lwm2m_reset_message(msg, true);
 		}
 
+		clear_ongoing_block_tx();
```

Comment:
What if there are multiple parallel block-wise transfers in progress. Aren't they all cleared here?

Reply:
This is intentional.
I don't believe any server would do parallel block-wise transfers with a single client.

I cannot be sure when the server is finished with the blockwise GET because I cannot keep any timeout. Server is supposed to retry. So I drop all ongoing block transfers when I receive a request that is not part of any ongoing operation.

Reply:
It bugs me whether we should differentiate between client-initiated BLOCK1 transfers (LwM2M Send) and server-initiated BLOCK2 transfers (LwM2M Get/Fetch). I think there is no ambiguity between those two. And right now it seems that we'd abort an ongoing block Send on reception of **any** (even non-block) request/response from the server. And conversely, the application can hijack any ongoing BLOCK2 transfer when calling `lwm2m_send_cb()`.

OFC we can agree on such a limitation, though it doesn't seem too difficult to avoid that.